### ISSUE TITLE
Not connect wireguard config with domain address.

### ISSUE THREAD

**Issue by Amir2367** [xchacha20-poly1305/husi#81]
### Version

```text
0.6.8
```

### Description

It does not connect to the Wireguard configuration with the domain address and the app crashes, and we have to completely close the app and re-enter the app.
You can see this problem in the video below

https://github.com/xchacha20-poly1305/husi/assets/83357422/bdae871e-47d4-4f6e-afe9-ae4c5183d2e9

### Reproduction

It does not connect to the Wireguard configuration with the domain address and the app crashes, and we have to completely close the app and re-enter the app.

### Integrity requirements

- [X] I confirm that my problem is the bug of husi, but not the problem of my own network environment or my server.
- [X] I confirm that I have provided the simplest configuration that can be used to reproduce the error I reported.
- [ ] I confirm that I have provided the complete configuration files and logs, rather than just providing parts I think are useful out of confidence in my own intelligence.

**Comment by xchacha20-poly1305** [comment]
Try [v0.7.0-beta.0](https://github.com/xchacha20-poly1305/husi/releases/tag/v0.7.0-beta.0)

**Comment by Amir2367** [comment]
> Try [v0.7.0-beta.0](https://github.com/xchacha20-poly1305/husi/releases/tag/v0.7.0-beta.0)

Yes, this problem is solved in the new version.

**Comment by Amir2367** [comment]
> Try [v0.7.0-beta.0](https://github.com/xchacha20-poly1305/husi/releases/tag/v0.7.0-beta.0)

No, this problem is not solved and the program crashes

https://github.com/xchacha20-poly1305/husi/assets/83357422/69426b64-00b5-410d-8adc-25740642bf03

**Comment by Mahdi-zarei** [comment]
I was not able to reproduce the issue, can you please provide logs when the connect button is non-responsive? Also please share your dns settings.

**Comment by Amir2367** [comment]
> I was not able to reproduce the issue, can you please provide logs when the connect button is non-responsive? Also please share your dns settings.

Yes, this problem is solved. But the problem of combining 2 wireguards is still there. It's like, when you combine 2 wireguards, it works at the first entry of the program. But when you disconnect, it doesn't connect anymore. And no engineering is done.
And you have to force stop the program and enter the program again and connect.
